### PR TITLE
[Bug] sql block rule npe

### LINKED ISSUE #9749: [Bug] SqlBlockRule NPE Exception
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

0.15
image
image

### What's Wrong?

We need judge sql whether exist.

### What You Expected?

metadata and matchSql can run success.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/incubator-doris/issues/9749

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/suites/sql_block_rule/test_sql_block_rule.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    qt_select """
+                SHOW SQL_BLOCK_RULE
+              """
+
+    sql """
+                DROP SQL_BLOCK_RULE test_rule_num
+              """
+
```
Comment (carlvinhust2012) [inline regression-test/suites/sql_block_rule/test_sql_block_rule.groovy:54]: Add a normal case maybe better.

File: fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRule.java [hunk 2]
```diff
@@ -33,12 +34,13 @@
...
 import java.util.regex.Pattern;
 
+/**
+ * Use for block some sql by rule.
+ **/
 public class SqlBlockRule implements Writable {
 
     public static final String NAME_TYPE = "SQL BLOCK RULE NAME";
 
-    public static final String DEFAULT_USER = "default";
```
Comment (carlvinhust2012) [inline fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRule.java:40]: why delete this line?

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/blockrule/SqlBlockRule.java:40]: DEFAULT_USER is used only when the first designed, and then defined with global, it is no longer needed

File: regression-test/suites/sql_block_rule/test_sql_block_rule.groovy (hunk 1 above)
Comment (stalary) [reply regression-test/suites/sql_block_rule/test_sql_block_rule.groovy:54]: Ok, I'm going to add it.

**Review by hf200012 — DISMISSED** [review]
LGTM

File: regression-test/conf/regression-conf.groovy [hunk 3]
```diff
@@ -28,7 +28,7 @@ feHttpAddress = "127.0.0.1:8030"
 feHttpUser = "root"
 feHttpPassword = ""
 
-beHttpAddress = "127.0.0.1:8040"
+beHttpAddress = "192.168.120.12:8140"
```
Comment (morningman) [inline regression-test/conf/regression-conf.groovy:31]: don't change this

**Review by morningman — APPROVED** [review]

**Pull request merged**